### DISCUSSION TITLE
Number of evaluated expressions (parameters)

### DISCUSSION THREAD

**Discussion by smeznar** [MilesCranmer/PySR#346] [General]
Hello,
I am comparing the performance of different symbolic regression approaches based on the number of evaluated expressions (Max 100,000). Could someone help me to understand which parameters should I set (and maybe which parameters are the most important) to constrain the approach to max 100.000 evaluations? 
I presume I must change parameters: niterations, populations, population_size, and ncyclesperiteration, but I am not sure how
Thank you in advance,
Sebastian

**Comment by MilesCranmer** [comment]
There is the `max_evals` parameter to set a max evaluations.

Note that the library has been tuned for performance in terms of *wall time*, rather than efficiency in terms of evaluations. So taking the default parameters will be quite suboptimal for evaluation efficiency. Sadly I’m not sure which parameters are most relevant for such an performance metric.

**Reply by smeznar** [reply]
Thank you for the quick answer, I overlooked that parameter. I'll keep the wall time thing in mind.

**Reply by MilesCranmer** [reply]
Btw, you can also stop by a specific time, or loss. The stopping criteria parameters are given here: https://astroautomata.com/PySR/api/#stopping-criteria
